src/mbgl/util/string_indexer.cpp

```diff
@@ -36,13 +39,25 @@ StringIdentity StringIndexer::get(std::string_view string) {
 
         if (const auto it = stringToIdentity.find(string); it == stringToIdentity.end()) {
             // this writer to insert
-            const StringIdentity id = identityToString.size();
-            identityToString.push_back(std::string(string));
-
-            [[maybe_unused]] auto result = stringToIdentity.insert(
-                {std::string_view(identityToString.back().data(), identityToString.back().length()), id});
-            assert(result.second);
+            auto& buffer = instance().buffer;
+            const auto previousCapacity = buffer.capacity();
 
+            const StringIdentity id = identityToString.size();
+            identityToString.push_back(buffer.size());
+            buffer.insert(buffer.end(), string.begin(), string.end());
+            buffer.push_back(0);
+            if (buffer.capacity() != previousCapacity) {
+                // reallocation happened
```

Comment:
How often do we actually see reallocation happening here?

Reply:
not that often, as `std::vector`'s default strategy afaik is exponential growth.
we could tune the `initialCapacity` so that it never happens, depending on the amount and length of strings to be indexed.